admin-dev/themes/new-theme/js/components/navbar-form-error-handler.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import NavbarHandler from './navbar-handler';
+
+type NavbarFormErrorHandlerType = {
+  form: HTMLElement;
+  navbarHandler: NavbarHandler;
+}
+
+/**
+ * This component is used as a wrapper for the NavbahHandler component. It allows handling
+ * tab redirection to the tab that contains some HTML form errors.
+ * Use this component only if you are using a form with NavbarHandler.
+ */
+export default class NavbarFormErrorHandler {
+  private readonly form: HTMLElement;
+
+  private readonly navbarHandler: NavbarHandler;
+
+  private firstInvalidField: HTMLElement | null = null;
+
+  constructor(options: NavbarFormErrorHandlerType) {
+    this.navbarHandler = options.navbarHandler;
+    this.form = options.form;
+
+    this.initListener();
+    this.resetInvalidField();
+  }
+
+  private findRequiredFieldsFromForm(): NodeListOf<HTMLElement> {
+    return this.form.querySelectorAll('[required]');
```

Comment:
I already made a similar suggestion previously, but are we sure we only want to check at `required` inputs?

That's the type of constraint that is bothering us for carrier page but there could be other use cases like `min/max` values, so to avoid having to update this component sometime soon why not checking all the inputs in the form instead?

Reply:
Yes, my bad 🙏🏻 I forgot to handle this one. Now, all fields in the form are selected in order to handle fields like the min/max attribute 🙂